zap/src/main/java/org/zaproxy/zap/extension/script/ScriptParam.java

```diff
@@ -46,6 +46,7 @@ public class ScriptParam extends AbstractParam {
     private static final String SCRIPT_ENABLED_KEY = "enabled";
     private static final String SCRIPT_DIRS = "dirs";
     private static final String SCRIPT_CONFIRM_REMOVE_DIR = "confRemdir";
+    private static final String SCRIPT_ENABLE_SCRIPTS_FROM_DIRS = "enableScriptsFromDirs";
```

Comment:
I think  SCRIPTS_BASE_KEY is only used for individual script configs. I thought following the same path as SCRIPT_DIRS would be more reasonable, so now SCRIPT_ENABLE_SCRIPTS_FROM_DIRS and SCRIPT_DIRS are on the same level
![2022-11-29_19-00](https://user-images.githubusercontent.com/103114290/204564023-ab488bed-0aee-4897-b3c4-45d933d83282.png)

Reply:
Not really, the other two keys are wrong as well, they should be all under the base `script` element.

Reply:
In that case, yes. It would make more sense to put them all under scripts element. But I think that  will be a really big breaking change. People have used  `-confg dirs` for as long as it existed.

Reply:
That's not a problem, we can migrate the old keys as needed. I'm not asking you to fix the other keys though, just add the new one in the right place but if you want to fix them, great :)

Reply:
From what I see, it's only a matter of 
```diff
diff --git a/zap/src/main/java/org/zaproxy/zap/extension/script/ScriptParam.java b/zap/src/main/java/org/zaproxy/zap/extension/script/ScriptParam.java
index a7a8a5484..c798de1d5 100644
--- a/zap/src/main/java/org/zaproxy/zap/extension/script/ScriptParam.java
+++ b/zap/src/main/java/org/zaproxy/zap/extension/script/ScriptParam.java
@@ -44,9 +44,9 @@ public class ScriptParam extends AbstractParam {
     private static final String SCRIPT_TYPE_KEY = "type";
     private static final String SCRIPT_FILE_KEY = "file";
     private static final String SCRIPT_ENABLED_KEY = "enabled";
-    private static final String SCRIPT_DIRS = "dirs";
-    private static final String SCRIPT_CONFIRM_REMOVE_DIR = "confRemdir";
-    private static final String SCRIPT_ENABLE_SCRIPTS_FROM_DIRS = "enableScriptsFromDirs";
+    private static final String SCRIPT_DIRS = SCRIPTS_BASE_KEY + ".dirs";
+    private static final String SCRIPT_CONFIRM_REMOVE_DIR = SCRIPTS_BASE_KEY + ".confRemdir";
+    private static final String SCRIPT_ENABLE_SCRIPTS_FROM_DIRS = SCRIPTS_BASE_KEY + ".enableScriptsFromDirs";
 
     private static final Logger logger = LogManager.getLogger(ScriptParam.class);
 
```
Tested, seems to work okay.

Reply:
That's not enough, you would also have to do something like: https://github.com/zaproxy/zaproxy/pull/7599/files#diff-83e062c0df1e7f4b18030a796521e134c4f6387cf8aad53444d05fce7f10533aR314

Reply:
Done.

Reply:
It still needs the tests.

Reply:
That the old keys are properly migrated to the new ones, e.g. `ScannerParamUnitTest#shouldMigrateOldOptions`. Check also with multiple `dirs` (it works fine but tests should still show it).

Reply:
```java
    @Test
    void shouldMigrateOldOptions() {
        // Given
        File dir1 = new File("/path/to/dir1");
        File dir2 = new File("/path/to/dir2");
        List<File> dirs = new ArrayList<>();
        dirs.add(dir1);
        dirs.add(dir2);
        configuration.setProperty("script.confRemdir", true);
        configuration.setProperty("script.dirs", dirs);
        // When
        param.load(configuration);
        // Then
        assertThat(param.isConfirmRemoveDir(), is(equalTo(true)));
        assertThat(param.getScriptDirs(), is(equalTo(dirs)));
        assertNull(configuration.getProperty("dirs"));
        assertNull(configuration.getProperty("confRemdir"));
    }
}
```
Something like this should work right? I'm really inexperienced with unit testing.

Reply:
cant get `assertThat(param.getScriptDirs(), is(equalTo(dirs)));` to work. `param.getScriptDirs()` seems to return a null list, but it shouldnt.

Reply:
Okay, I got it working, had to pass a string list to `setProperty`, not File list for `dirs`.  Good enough?
``` java
  @Test
  void shouldMigrateOldOptions() {
    // Given
    File dir1 = new File("/path/to/dir1");
    File dir2 = new File("/path/to/dir2");
    List<File> dirs = new ArrayList<>();
    dirs.add(dir1);
    dirs.add(dir2);
    
    configuration.setProperty("confRemdir", true);
    configuration.setProperty("dirs", new String[]{dir1.toString(), dir2.toString()});
    // When
    param.load(configuration);
    // Then
    assertThat(param.isConfirmRemoveDir(), is(equalTo(true)));
    assertThat(param.getScriptDirs(), is(equalTo(dirs)));
    assertNull(configuration.getProperty("dirs"));
    assertNull(configuration.getProperty("confRemdir"));
  }
```

Reply:
Don't know what actual paths to put into the commit to be honest.

Reply:
Bump?